taichi/backends/cuda/cuda_device.cpp

```diff
@@ -38,6 +38,8 @@ DeviceAllocation CudaDevice::allocate_memory_runtime(const AllocParams &params,
   AllocInfo info;
   if (params.host_read || params.host_write) {
     TI_NOT_IMPLEMENTED
+  } else if (params.use_cached && ccalloc.find_block(params.size)) {
```

Comment:
IMHO an easier logic here is 
- if use_cached, ccalloc.allocate() // could reuse or allocate new memory
- else: call allocate_llvm_runtime_memory_jit